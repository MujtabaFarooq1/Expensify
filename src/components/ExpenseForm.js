import React from "react";
import moment from "moment";
import { SingleDatePicker } from "react-dates";

export default class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      description: props.expense ? props.expense.description : "",
      note: props.expense ? props.expense.note : "",
      amount: props.expense ? (props.expense.amount / 100).toString() : "",
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      callenderFocused: false,
      error: "",
    };
  }

  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };
  onNoteChange = (e) => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };
  onAmountChange = (e) => {
    const amount = e.target.value;
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  };
  onDateChange = (createdAt) => {
    if (createdAt) {
      this.setState(() => ({ createdAt }));
    }
  };
  onFocusChange = ({ focused }) => {
    this.setState(() => ({ callenderFocused: focused }));
  };
  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.description || !this.state.amount) {
      // Set some error message
      this.setState(() => ({
        error:
          "Description and amount are required fields kindly fill in to continue",
      }));
    } else {
      // Clear The Error
      this.setState(() => ({ error: "" }));
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note,
      });
    }
  };
  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmit}>
          <input
            value={this.state.description}
            onChange={this.onDescriptionChange}
            type="text"
            placeholder="Description"
            autoFocus
          />
          <input
            value={this.state.amount}
            onChange={this.onAmountChange}
            type="number"
            placeholder="Amount"
          />
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.callenderFocused}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
          />
          <textarea
            value={this.state.note}
            placeholder="Add a note for your expense"
            onChange={this.onNoteChange}
          ></textarea>
          <button>Add Expense</button>
        </form>
      </div>
    );
  }
}
