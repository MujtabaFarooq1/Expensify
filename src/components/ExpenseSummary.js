import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import { Link } from "react-router-dom";
import selectExpensesTotal from "../selectors/expense-total";
import selectExpenses from "../selectors/expenses";

export const ExpenseSummary = ({ expensesTotal, expensesCount }) => {
  const expensesWord = expensesCount === 1 ? " expense " : " expenses ";
  const formattedExpensesTotal = numeral(expensesTotal / 100).format(
    " $0,0.00"
  );
  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">
          Viewing <span>{expensesCount}</span> {expensesWord} totalling{" "}
          <span>{formattedExpensesTotal}</span>
        </h1>
        <div className="page-header__actions">
          <Link className="btn btn--primary" to="/create">
            Add Expense
          </Link>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);

  return {
    expensesCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses),
  };
};

export default connect(mapStateToProps)(ExpenseSummary);
