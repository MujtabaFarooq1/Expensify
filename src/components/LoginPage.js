import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";
// import {firebase} from "../actions/"

export const LoginPage = ({ startLogin }) => (
  <div>
    <button type="submit" onClick={startLogin}>
      Login
    </button>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
