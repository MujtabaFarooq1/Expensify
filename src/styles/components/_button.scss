.btn {
  font-size: $font-size-large;
  font-size: $m-size;
  padding: $s-size;
  font-weight: 300;
  text-transform: capitalize;
  color: #fff;
  border-radius: 0.3rem;
  border: none;
  text-decoration: none;
  display: inline-block;
  line-height: 1;

  &--primary {
    background-color: $primary-blue;
  }

  &--secondary {
    background-color: $medium-grey;
  }

  &--outlined {
    background-color: transparent;
    border: 1px solid transparent;
    transition: all 0.2s ease-in-out;

    &:hover {
      border-color: #fff;
    }
  }
}
