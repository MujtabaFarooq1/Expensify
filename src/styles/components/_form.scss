.form {
  //
  display: flex;
  flex-direction: column;

  > * {
    margin-bottom: $m-size;
  }

  &__error {
    margin: 0 0 $m-size 0;
    font-style: italic;
    color: crimson;
    font-size: $font-size-large;
    padding: $s-size;
    outline: 1px solid crimson;
    text-transform: capitalize;
  }
}
