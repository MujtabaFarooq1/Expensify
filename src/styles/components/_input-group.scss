.input-group {
  display: flex;
  flex-direction: column;
  margin-bottom: $m-size;

  @media (min-width: $desktop-breakpoint) {
    margin-bottom: $l-size;
    flex-direction: row;
  }

  &__item {
    display: block;
    margin-bottom: $s-size;

    @media (min-width: $desktop-breakpoint) {
      &:not(:last-child) {
        margin: 0 $s-size 0 0;
      }
    }
  }
}
