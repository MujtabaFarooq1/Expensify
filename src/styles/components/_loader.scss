.loader {
  height: 100vh;
  width: 100vw;
  display: flex;
  align-items: center;
  justify-content: center;
  background: $light-grey;
  &__img {
    height: 6rem;
    width: 6rem;
  }
}
