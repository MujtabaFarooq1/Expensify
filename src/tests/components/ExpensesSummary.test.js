import React from "react";
import { shallow } from "enzyme";
import { ExpenseSummary } from "../../components/ExpenseSummary";
import expenses from "../fixtures/expenses";

test("should render summary with one expense", () => {
  const wrapper = shallow(
    <ExpenseSummary expensesCount={1} expensesTotal={expenses[1].amount} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should render summary with more or no expenses", () => {
  const wrapper = shallow(
    <ExpenseSummary
      expensesCount={3}
      expensesTotal={
        expenses[0].amount + expenses[1].amount + expenses[2].amount
      }
    />
  );
  expect(wrapper).toMatchSnapshot();
});
