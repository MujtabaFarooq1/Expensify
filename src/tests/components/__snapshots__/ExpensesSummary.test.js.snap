// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render summary with more or no expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
     expenses 
     totalling
     
     $64.61
  </h1>
</div>
`;

exports[`should render summary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
     expense 
     totalling
     
     $10.87
  </h1>
</div>
`;
